### ISSUE TITLE
Syntax issue with a zero-argument pure function

### ISSUE THREAD

**Issue by otrack** [informalsystems/quint#881]
I am not sure to understand why this is failing:
```
module A {
  pure def foo(): bool = {
    true
  }
  var x: bool
  action init = all {
    x' = foo()
  }
}
```
as follows:
```
static analysis error: error: [QNT000] Can't unify different types of effects
Trying to unify Pure and () => e0
Trying to infer effect for operator application in foo()
```
whereas the code below actually works:
```
module A {
  pure def foo(): bool = {
    true
  }
  var x: bool
  action init = all {
    x' = foo
  }
}
```

**Comment by konnov** [comment]
I think our intent was to make people use `val` in this case, not `def`. The error message is confusing though. @bugarela wdyt?

**Comment by bugarela** [comment]
I can't think of a scenario where someone would want to used a zero-argument `def` instead of a `val`. So perhaps we should drop support for zero argument `def`s at all?

**Comment by konnov** [comment]
> I can't think of a scenario where someone would want to used a zero-argument `def` instead of a `val`. So perhaps we should drop support for zero argument `def`s at all?

I agree, in case of `def`s it does not make sense to write something like `def foo() = ...`. However, it could be good for actions, as discussed in #220.
